### PR TITLE
feat: Add hover styles to TOC entries

### LINKED ISSUE #9016: Add hover styles to TOC entries
### Describe the problem

On hover, there is no indication which element in the TOC you are interacting with.

### Expected fix

Add hover styles, similar to how we did for the main sidenav.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ulisseshen**
_Description of what this PR is changing or adding, and why:_
Was added hover style to TOC entries 
_Issues fixed by this PR (if any):_
Closes #9016 
## Presubmit checklist

- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/_sass/components/_toc.scss [hunk 1]
```diff
@@ -4,6 +4,15 @@
   display: none;
   right: 0;
 
+
+  @mixin nav-link-not-active-hover {
+    &:not(.disabled) {
```
Comment (parlough) [inline src/_sass/components/_toc.scss:9]: I'm not sure the TOC entries can be disabled, at least I can't find any area where they can be. Do you know of any? If not, we can likely remove the `&:not(.disabled) {` portion.

File: src/_sass/components/_toc.scss [hunk 2]
```diff
@@ -4,6 +4,15 @@
   display: none;
   right: 0;
 
+
+  @mixin nav-link-not-active-hover {
```
Comment (parlough) [inline src/_sass/components/_toc.scss:8]: Since this doesn't have a special case for active, perhaps just `nav-link-hover`?

**Review by parlough — COMMENTED** [review]
Thanks so much for this accessibility improvement!

A question and a suggestion:

File: src/_sass/components/_toc.scss (hunk 1 above)
Comment (ulisseshen) [reply src/_sass/components/_toc.scss:9]: You is correctly, I only copy and paste from another place.

File: src/_sass/components/_toc.scss (hunk 2 above)
Comment (ulisseshen) [reply src/_sass/components/_toc.scss:8]: This name is more clean.

**Review by parlough — APPROVED** [review]
Nice 🥳 ! Thanks so much :D

**Pull request merged**